### PR TITLE
fix: `DefaultResolveFn` should handle with `map[K]V` for `K` that has `string` as the underlying type

### PR DESCRIPTION
**Pull request by bkrukowski**
The current implementation returns the following errors for maps with a custom type for keys:

```
reflect.Value.MapIndex: value of type string is not assignable to type .*
```

### CODE REVIEW

**Comment by bkrukowski** [comment]
@chris-ramon could you please merge that? it fixes a bug

**Comment by chris-ramon** [comment]
Hi @bkrukowski - thanks for sending a PR.

I gave a quick look, the change is not needed, since the end result of `val` is the same `reflect.Value`:

```go
val := r.MapIndex(reflect.ValueOf(p.Info.FieldName).Convert(r.Type().Key()))
```

Could you provide an example of the following error you shared above ? - That is related to the PR ?

```go
reflect.Value.MapIndex: value of type string is not assignable to type .*
```

**Comment by bkrukowski** [comment]
@chris-ramon please take a look, I removed that code without removing the newly added test, and it fails:

image

https://app.circleci.com/pipelines/github/graphql-go/graphql/128/workflows/41b52c89-f893-43a4-a0d4-709dbd0c6fac/jobs/1626

The following code creates a new value of the `string` type:

```go
reflect.ValueOf(p.Info.FieldName)
```
It causes an issue for maps that have keys of a custom, and the underlying type is `string`, an example in the tests:

```go
type (
	CustomMap map[string]interface{}

	CustomKey  string
	CustomMap2 map[CustomKey]interface{}
)

// ...

	scenarios := []struct {
		query    string
		data     interface{}
		expected interface{}
	}{
		{ // it works
			query: query,
			data: CustomMap{
				"a": "1",
				"b": "2",
			},
			expected: map[string]interface{}{
				"data": map[string]interface{}{
					"a": "1",
				},
			},
		},
		{ // it does not work without the suggested patch
			query: query,
			data: CustomMap2{
				"a": "1",
				"b": "2",
			},
			expected: map[string]interface{}{
				"data": map[string]interface{}{
					"a": "1",
				},
			},
		},
	}
```

**Comment by chris-ramon** [comment]
@bkrukowski as explained before in the comment: https://github.com/graphql-go/graphql/pull/697#issuecomment-2366893547 - The change is not needed, going ahead and closing the PR.

**Comment by bkrukowski** [comment]
@chris-ramon kindly please take a look

### TL;DR

The given code panics here, and I think you missed that point.

### Full story

> @bkrukowski as explained before in the comment: [#697 (comment)](https://github.com/graphql-go/graphql/pull/697#issuecomment-2366893547) - The change is not needed, going ahead and closing the PR.

it's incorrect

> I gave a quick look, the change is not needed, since the end result of `val` is the same as `reflect.Value`:
> 
> ```go
> val := r.MapIndex(reflect.ValueOf(p.Info.FieldName).Convert(r.Type().Key()))
> ```

No, it isn't. Without the suggested fix that code **panics** because we pass a wrong argument to `MapIndex`, so the `val` does not exist. Example:

```go
package main

import (
	"github.com/graphql-go/graphql"
)

type MyKey string

type MyMap map[MyKey]any

func main() {
	myMap := MyMap{"key": "value"}

	graphql.DefaultResolveFn(graphql.ResolveParams{Source: myMap})
}
```

Output:

```
panic: reflect.Value.MapIndex: value of type string is not assignable to type main.MyKey

goroutine 1 [running]:
reflect.Value.assignTo({0x528da0?, 0x681160?, 0x10?}, {0x555582, 0x16}, 0x528be0, 0x0)
	/usr/local/go-faketime/src/reflect/value.go:3358 +0x299
reflect.Value.MapIndex({0x530fe0?, 0xc0000b0ea0?, 0x41103b?}, {0x528da0, 0x681160, 0x98})
	/usr/local/go-faketime/src/reflect/value.go:1805 +0xe8
github.com/graphql-go/graphql.DefaultResolveFn({{0x530fe0, 0xc0000b0ea0}, 0x0, {{0x0, 0x0}, {0x0, 0x0, 0x0}, 0x0, {0x0, ...}, ...}, ...})
	/tmp/gopath2667206492/pkg/mod/github.com/graphql-go/graphql@v0.8.1/executor.go:998 +0x365
main.main()
	/tmp/sandbox4170389645/prog.go:16 +0xb0
```

https://go.dev/play/p/2LiyVcmhqUq

Since that library recovers all the panics, it returns the following error eventually:

```
reflect.Value.MapIndex: value of type string is not assignable to type main.MyKey
```
